### DISCUSSION TITLE
Security considerations when exposing Minio on a public URL

### DISCUSSION THREAD

**Discussion by binaryfire** [minio/minio#18887] [Q&A]
Hi all,

Normally I put my services behind a reverse proxy like Cloudflare but they have a 100mb upload limit and there are issues with serving certain file types. So I've just added certs to my MinIO instance, set `--address :443` and exposed it on a public URL (the console is still on 9001 and is ip-restricted).

I have some buckets which are public and are served directly from Minio (web app media etc) and others which are private. I'm wondering if there are any additional security precautions I should be taking when exposing MinIO directly like this? I.e. things like IP-based rate limiting, DDoS protection etc. Or do these things not pose the same kind of problem for MinIO as they do for regular web apps?

Thanks!

**Comment by klauspost** [comment]
You should apply the same care as regular services. There is no DDOS protection, so if you expect that you should mitigate against it with the regular tools you'd use for that.

**Reply by binaryfire** [reply]
I'm not expecting a DDoS attack, I'm curious about what the majority of deployments look like. Do most people put an additional HTTP layer in front of MinIO instances that are serving public files?

Most of the videos I've seen expose it directly to the internet eg. this one from your channel: https://youtu.be/d48yrp978aA?si=mwEYBmWW6odIckuy&t=224.

**Reply by klauspost** [reply]
Most professional installs are being a load balancer/reverse proxy since they are distributed setups. 

We offer setup assistance in our [subscriptions](https://min.io/pricing) if you are looking to do a commercial setup. Otherwise just apply regular protection you'd put against any service you expose.
